core/src/main/java/com/datastrato/gravitino/storage/relational/mapper/CatalogMetaMapper.java

```diff
@@ -149,14 +149,14 @@ Integer updateCatalogMeta(
   @Update(
       "UPDATE "
           + TABLE_NAME
-          + " SET deleted_at = UNIX_TIMESTAMP(CURRENT_TIMESTAMP(3)) * 1000.0"
+          + " SET deleted_at =(UNIX_TIMESTAMP() * 1000.0) + EXTRACT(MICROSECOND FROM CURRENT_TIMESTAMP(3)) / 1000"
```

Comment:
Don't remove the whitespace here. Also can you please check it this `EXTRACT(MICROSECOND FROM CURRENT_TIMESTAMP(3)` is supported by other databases.

Reply:
I have checked:  `EXTRACT(MICROSECOND FROM CURRENT_TIMESTAMP(3)` does not work in PG, maybe we need to change the SQL accordingly.

The reason why I modified here is that for H2,  `UNIX_TIMESTAMP(CURRENT_TIMESTAMP(3)) * 1000.0`  returns a full second time instead of a millisecond time, which is quite different from MySQL, so I change it.

Reply:
If this sql dialect only supported by for H2, MySQL, how do you handle other embedded database or other databases like PG? You will definitely have a chance to meet microsecond issues even not for embedded database.

Reply:
> You will definitely have a chance to meet microsecond issues even for embedded database.

Yeah, I would be another issue. I need these changes to make SQL between H2 and MySQL work now. When It comes to other dialects like PG or Oracle, we need to design a mechanism that can be adapted to several different databases. This seems to be beyond the scope of this PR.